rust/src/domains/polars/expr/mod.rs

```diff
@@ -100,14 +100,23 @@ impl<C: Context> Debug for ExprDomain<C> {
 
 pub trait ExprMetric<C>: Metric {
     type InnerMetric: Metric<Distance = Self::Distance>;
+    fn inner_metric(&self) -> Self::InnerMetric;
```

Comment:
This change should be re-committed in the `ExprDomain` PR (when you restack it will disappear here).